### PR TITLE
feat: algokit network support

### PR DESCRIPTION
**Pull request by negar-abbasi**

### CODE REVIEW

File: template_content/smart_contracts/{% if deployment_language == 'python' %}config.py{% endif %} [hunk 1]
```diff
@@ -42,11 +43,11 @@ def deploy(
...
                 on_schema_break=(
-                    OnSchemaBreak.ReplaceApp if is_local else OnSchemaBreak.Fail
+                    OnSchemaBreak.AppendApp if is_main else OnSchemaBreak.ReplaceApp
                 ),
-                on_update=OnUpdate.UpdateApp if is_local else OnUpdate.Fail,
-                allow_delete=is_local,
-                allow_update=is_local,
+                on_update=OnUpdate.AppendApp if is_main else OnUpdate.UpdateApp,
+                allow_delete=is_main,
+                allow_update=is_main,
```
Comment (achidlow) [inline template_content/smart_contracts/{% if deployment_language == 'python' %}config.py{% endif %}:50]: I think you want `not is_main` here for `allow_delete` and `allow_update`?

File: template_content/smart_contracts/{% if deployment_language == 'typescript' %}deploy-config.ts{% endif %} [hunk 2]
```diff
@@ -32,10 +32,10 @@ export async function deploy(name: (typeof contracts)[number], appSpec: AppSpec)
     // Edit this to add the custom deployment logic for each contract
     case 'HelloWorldApp':
       const app = await appClient.deploy({
-        allowDelete: isLocal,
-        allowUpdate: isLocal,
-        onSchemaBreak: isLocal ? 'replace' : 'fail',
-        onUpdate: isLocal ? 'update' : 'fail',
+        allowDelete: isMain,
+        allowUpdate: isMain,
```
Comment (achidlow) [inline template_content/smart_contracts/{% if deployment_language == 'typescript' %}deploy-config.ts{% endif %}:36]: similarly here, I think these two conditions should be inverted yeah?

Comment (daniel-makerx) [reply template_content/smart_contracts/{% if deployment_language == 'typescript' %}deploy-config.ts{% endif %}:36]: That's correct, allowDelete and allowUpdate should be `!isMain` and likewise for the Python deployment code

File: template_content/smart_contracts/{% if deployment_language == 'python' %}config.py{% endif %} (hunk 1 above)
Comment (negar-abbasi) [reply template_content/smart_contracts/{% if deployment_language == 'python' %}config.py{% endif %}:50]: LocalNet & TestNet
- on_schema_break = OnSchemaBreak.ReplaceApp
- on_update = OnUpdate.UpdateApp

MainNet
- on_schema_break = OnSchemaBreak.AppendApp
- on_update = OnSchemaBreak.AppendApp

we want append for mainnet right?

File: template_content/algokit.toml
Comment (daniel-makerx) [inline template_content/algokit.toml:1]: This file shouldn't be renamed

**Review by daniel-makerx — APPROVED** [review]

File: template_content/smart_contracts/{% if deployment_language == 'typescript' %}deploy-config.ts{% endif %} [hunk 3]
```diff
@@ -17,7 +17,7 @@ export async function deploy(name: (typeof contracts)[number], appSpec: AppSpec)
     },
     algod,
   )
-  const isLocal = await algokit.isLocalNet(algod)
+  const isMain = await algokit.isMainNet(algod)
```
Comment (daniel-makerx) [inline template_content/smart_contracts/{% if deployment_language == 'typescript' %}deploy-config.ts{% endif %}:20]: Just to make this clearer, can we change this to `isMainNet`

File: template_content/smart_contracts/{% if deployment_language == 'python' %}config.py{% endif %} [hunk 4]
```diff
@@ -21,14 +21,15 @@ logger = logging.getLogger(__name__)
...
+
 # define deployment behaviour based on supplied app spec
 def deploy(
     algod_client: AlgodClient,
     indexer_client: IndexerClient,
     app_spec: ApplicationSpecification,
     deployer: Account,
 ) -> None:
-    is_local = is_localnet(algod_client)
+    is_main = is_mainnet(algod_client)
```
Comment (daniel-makerx) [inline template_content/smart_contracts/{% if deployment_language == 'python' %}config.py{% endif %}:32]: Also change this to `is_mainnet`, will need to change the import as well and use `algokit_utils.is_mainnet`

**Pull request merged**